src/Umbraco.Web/Cache/PageCacheRefresher.cs

```diff
@@ -75,23 +84,36 @@ public override void Refresh(int id)
         public override void Remove(int id)
         {
             if (Suspendable.PageCacheRefresher.CanUpdateDocumentCache)
+            {
                 content.Instance.ClearDocumentCache(id, false);
+                content.Instance.ClearPreviewXmlContent(id);
+            }
+                
             ClearCaches();
             base.Remove(id);
         }
 
         public override void Refresh(IContent instance)
         {
             if (Suspendable.PageCacheRefresher.CanUpdateDocumentCache)
-                content.Instance.UpdateDocumentCache(new Document(instance));
+            {
+                Document d = new Document(instance);
+                content.Instance.UpdateDocumentCache(d);
+                content.Instance.UpdatePreviewXmlContent(d);
+            }
+
             ClearCaches();
             base.Refresh(instance);
         }
 
         public override void Remove(IContent instance)
         {
             if (Suspendable.PageCacheRefresher.CanUpdateDocumentCache)
+            {               
                 content.Instance.ClearDocumentCache(new Document(instance), false);
+                content.Instance.ClearPreviewXmlContent(instance.Id);

```

Comment:
Any reason why `ClearDocumentCache` also doesn't use `instance.Id`, but creates a new instance of `Document`? Looks like the overload with the documentId does it a little bit safer: https://github.com/umbraco/Umbraco-CMS/blob/d11a1946f573f055c88a219bee3f7f1d6b058aa7/src/Umbraco.Web/umbraco.presentation/content.cs#L404-L420

Reply:
Don't know why. This was already in place, so I did not touch it.